library/Vanilla/Embeds/TwitchEmbed.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+
+/**
+ * Twitch Embed.
+ */
+class TwitchEmbed extends VideoEmbed {
+
+    const DEFAULT_HEIGHT = 300;
+
+    const DEFAULT_WIDTH = 400;
+
+    /** @inheritdoc */
+    protected $domains = ['www.twitch.tv', 'clips.twitch.tv'];
+
+    /*** @var string */
+    private $urlType;
+
+    /**
+     * TwitchEmbed constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct('twitch', 'video');
+        $this->urlType = '';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url)
+    {
+        $data = [];
+
+        if ($this->isNetworkEnabled()) {
+            $videoID = $this->getUrlInformation($url);
```

Comment:
As much of the url parsing logic should be outside of the networkingEnabled call, because the stuff inside of it never runs in a unit test. See the youtube embed for a better example.

 Additionally this `->getUrlInformation` function should be made public and have unit tests written for it.